omnibus/resources/agent/msi/cal/usercreate.cpp

```diff
@@ -315,9 +317,51 @@ int doesUserExist(MSIHANDLE hInstall, const CustomActionData& data)
         return 0;
     }
     if (ERROR_INSUFFICIENT_BUFFER != err) {
-        // we don't know what happened
-        WcaLog(LOGMSG_STANDARD, "doesUserExist: Lookup Account Name: Expected insufficient buffer, got error %d 0x%x", err, err);
-        return -1;
+        err = GetLastError();
+        if (!isDC) {
+            // can only try this if we're not on a primary/backup DC; on DCs we must
+            // be able to contact the domain authority.  
+            if (err >= ERROR_NO_TRUST_LSA_SECRET && err <= ERROR_TRUST_FAILURE) {
+                WcaLog(LOGMSG_STANDARD, "Can't reach domain controller %d", err);
+                // if the user specified a domain, then also must be able to contact
+                // the domain authority
+                if (data.getDomainPtr() == NULL) {
+                    WcaLog(LOGMSG_STANDARD, "trying fully qualified local account");
+                    bRet = LookupAccountName(NULL, data.getFullUsername().c_str(), newsid, &cbSid, refDomain, &cchRefDomain, &use);
+                    if (bRet) {
+                        // this should *never* happen, because we didn't pass in a buffer large enough for
+                        // the sid or the domain name.
+                        WcaLog(LOGMSG_STANDARD, "doesUserExist: Lookup Account Name: Unexpected error %d 0x%x", err, err);
+                        return -1;
+                    }
+                    err = GetLastError();
+                    if (ERROR_NONE_MAPPED == err) {
+                        // this user doesn't exist.  We're done
+                        WcaLog(LOGMSG_STANDARD, "retried user doesn't exist");
+                        return 0;
+                    }
+                    if (ERROR_INSUFFICIENT_BUFFER != err) {
+                        WcaLog(LOGMSG_STANDARD, "Failed retry of lookup account name %d", err);
+                        return -1;
+                    }
```

Comment:
If we get some other error, we are not logging it. Add an `else` here?

Reply:
any error that is not `ERROR_INSUFFICIENT_BUFFER` is already caught by the above `ERROR_INSUFFICIENT_BUFFER != err`